### PR TITLE
Feature/general fixes

### PR DESCRIPTION
**Pull request by AndreiOla**
**Changed**
I made this branch for splitting this PR: https://github.com/boomerang-io/flow.service.workflow/pull/210
This branch contains the not directly related changes to Eventing and I will make a different branch that will contain the Eventing changes.

### CODE REVIEW

**Comment by AndreiOla** [comment]
@tlawrie @amhudson Please make some time to review these changes. There's a risk to get more changes on main and it takes a lot of time to re sync.

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java [hunk 1]
```diff
@@ -28,18 +27,16 @@
 import javax.servlet.http.Part;
 import org.springframework.util.StreamUtils;
 
-/*
- * Reference: https://www.baeldung.com/spring-reading-httpservletrequest-multiple-times
- */
```
Comment (tlawrie) [inline src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:33]: Let's leave this in. We need to know the reference for where this method is duplicated from. it helps retain knowledge.

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java [hunk 2]
```diff
@@ -28,18 +27,16 @@
...
   private final byte[] buffer;
   private final HttpServletRequest request;
 
   public MultiReadHttpServletRequest(HttpServletRequest request) throws IOException {
     this.request = request;
-    InputStream requestInputStream = request.getInputStream();
-    this.buffer = StreamUtils.copyToByteArray(requestInputStream);
+    ByteArrayOutputStream bufferStream = new ByteArrayOutputStream();
+    StreamUtils.copy(request.getInputStream(), bufferStream);
+    this.buffer = bufferStream.toByteArray();
```
Comment (tlawrie) [inline src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:39]: Can you run me through this change?

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java [hunk 3]
```diff
@@ -325,6 +316,11 @@ public int getServerPort() {
     return request.getServerPort();
   }
 
+  @Override
+  public BufferedReader getReader() throws IOException {
+    return request.getReader();
+  }
+
   @Override
```
Comment (tlawrie) [inline src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:324]: Can you run me through this change?

**Review by tlawrie — COMMENTED** [review]
A few minor queries. Otherwise looks good.

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java (hunk 2 above)
Comment (AndreiOla) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:39]: I think this is done by @corcoja . As far as I understand it, there was a wrapper Type 'unboxed': InputStream. This is in relation with `getReader` method.

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java (hunk 3 above)
Comment (AndreiOla) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:324]: @corcoja ?

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java (hunk 1 above)
Comment (AndreiOla) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:33]: I will put this back.

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java (hunk 3 above)
Comment (corcoja) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:327]: @AndreiOla, I'm not entirely sure. Looking at GIT history, this is part of [this](https://github.com/boomerang-io/flow.service.workflow/commit/4dbb6508cc21aaa92d370555ecc33705b6fb8e89) commit, which was pushed by yourself.
https://github.com/boomerang-io/flow.service.workflow/commit/4dbb6508cc21aaa92d370555ecc33705b6fb8e89

Comment (corcoja) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:327]: This is a replacement for this code:
```java
  @Override
   public BufferedReader getReader() throws IOException {
     ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(this.buffer);
     return new BufferedReader(new InputStreamReader(byteArrayInputStream));
   }
```

File: src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java (hunk 2 above)
Comment (corcoja) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java:42]: Same comment as below 😅 This PR changes the code snippet below. Compared to the old code, it looks like a longer way of copying one array to another, but @AndreiOla should know better.

```java
    InputStream requestInputStream = request.getInputStream();
    this.buffer = StreamUtils.copyToByteArray(requestInputStream);
```

Comment (AndreiOla) [reply src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java]: Well, it seems I messed some things up, when I extracted the changes. I reverted them on how they were done by @tlawrie .

**Review by amhudson — APPROVED** [review]

**Comment by AndreiOla** [comment]
Hey @tlawrie Please review this one!

**Comment by tlawrie** [comment]
Merging IBM requested changes as approved by @amhudson.

**Pull request merged**